src/libsyntax/ext/base.rs

```diff
@@ -811,9 +809,8 @@ impl<'a> ExtCtxt<'a> {
     pub fn monotonic_expander<'b>(&'b mut self) -> expand::MacroExpander<'b, 'a> {
         expand::MacroExpander::new(self, true)
     }
-
-    pub fn new_parser_from_tts(&self, tts: &[tokenstream::TokenTree]) -> parser::Parser<'a> {
-        parse::stream_to_parser(self.parse_sess, tts.iter().cloned().collect(), MACRO_ARGUMENTS)
+    pub fn new_parser_from_token_stream(&self, stream: TokenStream) -> parser::Parser<'a> {
```

Comment:
Since the old function is removed, perhaps we can keep the old name `new_parser_from_tts`?
Token streams are conventionally named `tts` all the time.
(Then commits could be squashed into one.)